cachi2/core/package_managers/gomod.py

```diff
@@ -48,6 +48,9 @@ class _GolangModel(pydantic.BaseModel):
     """
 
     class Config:
+        # provides a __hash__ function, used for deduplicating modules
+        frozen = True
```

Comment:
Deduplicating GoModules by their hash probably won't work the way you expect - the same module is reported differently in `go mod download -json` vs. `go list -deps -json`

See https://github.com/containerbuildsystem/cachito/pull/853#discussion_r1170946248

You might want to de-duplicate by name+version instead, and take extra care if the original_name is relevant (beacuse `go mod download` doesn't report the original name)